AppCenter/AppCenterSDK+HiDrive.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  AppCenterSDK+SMFLogger.swift
+//  SmartMobileFactory
+//
+//  Created by Konstantin Deichmann on 11.09.19.
+//  Copyright © 2019 SmartMobileFactory. All rights reserved.
+//
+
+import Foundation
+import AppCenter
+import AppCenterCrashes
+import AppCenterDistribute
+import SMFLogger
+
+
+fileprivate enum AppCenterConstants {
+
+	static let appSecretKey					= "AppCenterAppSecret"
+	static let smfLogUploadMaxSize	: Int	= 5000
+}
+
+class AppCenterSDK: NSObject {
+
+	// MARK: - Private static properties
+
+	fileprivate static let isDebugBuild		: Bool = {
+		#if DEBUG
+		return true
+		#else
+		return false
+		#endif
+	}()
+
+	// MARK: - Private properties
+
+	fileprivate static var shared			: AppCenterSDK?
+	fileprivate var configuration			: Configuration?
+
+	// MARK: - Public properties
+
+	static var wasInitialized				: Bool {
+		return MSAppCenter.isConfigured()
+	}
+
+	// MARK: - Methods
+
+	/// This will setup the AppCenterSDK with the common base configuration. Crashes will be detected if the app is build with the release build type.
+	/// Distribution can be enabled using the configuration
+	///
+	/// - Parameters:
+	///   - configuration: Configuration object
+	static func setup(configuration: AppCenterSDK.Configuration = .default) {
+		guard (self.isDebugBuild == false || configuration.enableDebug == true) else {
+			return
+		}
+
+		let services = (configuration.isDistributionEnabled == true) ? [MSCrashes.self, MSDistribute.self] : [MSCrashes.self]
+
+		MSAppCenter.start(configuration.appSecret, withServices: services)
+		MSCrashes.setEnabled(configuration.enableCrashes)
+		MSCrashes.setDelegate(shared)
```

Comment:
shared is always nil.

Reply:
`self.shared = AppCenterSDK()` needs to be called before setting the MSCrashes delegate.